libraries/rush-lib/src/logic/operations/ConsoleTimelinePlugin.ts

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { ITerminal } from '@rushstack/node-core-library';
+import colors from 'colors/safe';
+import { IPhase } from '../../api/CommandLineConfiguration';
+import {
+  ICreateOperationsContext,
+  IPhasedCommandPlugin,
+  PhasedCommandHooks
+} from '../../pluginFramework/PhasedCommandHooks';
+import { IExecutionResult } from './IOperationExecutionResult';
+import { OperationStatus } from './OperationStatus';
+
+const PLUGIN_NAME: 'ConsoleTimelinePlugin' = 'ConsoleTimelinePlugin';
+
+/**
+ * Phased command plugin that emits a timeline to the console.
+ */
+export class ConsoleTimelinePlugin implements IPhasedCommandPlugin {
+  private readonly _terminal: ITerminal;
+
+  public constructor(terminal: ITerminal) {
+    this._terminal = terminal;
+  }
+
+  public apply(hooks: PhasedCommandHooks): void {
+    hooks.afterExecuteOperations.tap(
+      PLUGIN_NAME,
+      (result: IExecutionResult, context: ICreateOperationsContext): void => {
+        _printTimeline(this._terminal, result);
+      }
+    );
+  }
+}
+
+/**
+ * Timeline - a wider column width for printing the timeline summary
+ */
+const TIMELINE_WIDTH: number = 109;
+
+/**
+ * Timeline - symbols representing each operation status
+ */
+const TIMELINE_CHART_SYMBOLS: Record<OperationStatus, string> = {
+  [OperationStatus.Ready]: '?',
+  [OperationStatus.Executing]: '?',
+  [OperationStatus.Success]: '#',
+  [OperationStatus.SuccessWithWarning]: '!',
+  [OperationStatus.Failure]: '!',
+  [OperationStatus.Blocked]: '.',
+  [OperationStatus.Skipped]: '%',
+  [OperationStatus.FromCache]: '%',
+  [OperationStatus.NoOp]: '%'
+};
+
+/**
+ * Timeline - colorizer for each operation status
+ */
+const TIMELINE_CHART_COLORIZER: Record<OperationStatus, (string: string) => string> = {
+  [OperationStatus.Ready]: colors.yellow,
+  [OperationStatus.Executing]: colors.yellow,
+  [OperationStatus.Success]: colors.green,
+  [OperationStatus.SuccessWithWarning]: colors.yellow,
+  [OperationStatus.Failure]: colors.red,
+  [OperationStatus.Blocked]: colors.red,
+  [OperationStatus.Skipped]: colors.green,
+  [OperationStatus.FromCache]: colors.green,
+  [OperationStatus.NoOp]: colors.gray
+};
+
+interface ITimelineRecord {
+  startTime: number;
+  endTime: number;
+  durationString: string;
+  name: string;
+  status: OperationStatus;
+}
+
```

Comment:
It's worth having a comment with some example output text. Given that it's ASCII it should be easy to put an inline sample.